fakturoid/paging.py

```diff
@@ -1,27 +1,39 @@
+import itertools
 from itertools import islice
 
 from fakturoid import six
 
 
 class PagedResource(object):
-    """List adapter for paged resources. Returns sliceable lazy loaded object."""
+    """List adapter for paged resources. Do not use it unless required
+       as it loads all the pages."""
 
     def __init__(self):
+        self.objects = None
         self.pages = {}
-        self.page_size = None
-        self.page_count = None
+        self.item_count = None
 
     def load_page(self, n):
         raise NotImplementedError("You must implement load_page method.")
 
-    def ensure_paging_info(self):
-        if self.page_count is None or self.page_size is None:
-            # load any page to get page count and page size
-            self.get_page(0)
+    def load_all_pages(self):
+        self.objects = []
+        self.item_count = 0
+        for n in itertools.count(start=0):
+            try:
+                page = self.get_page(n)
+                self.objects.extend(page)
+                self.item_count += len(page)
+            except IndexError:
+                # we've reached the end of list
+                break
+        self.has_all_pages = True
```

Comment:
`self.has_all_pages` is not initialized in `__init__()`

Reply:
Thank you for the notes.

Updated.